File: src/test/java/com/mpe85/grampa/runner/ParseRunnerTest.java
package com.mpe85.grampa.runner;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

import com.mpe85.grampa.parser.Parser;
import com.mpe85.grampa.rule.impl.StringRule;

public class ParseRunnerTest {
	
	@Test
	public void test_run() {
		final Parser<Void> parser = () -> new StringRule<>("foo");
		final DefaultParseRunner<Void> runner = new DefaultParseRunner<>(parser);
		assertTrue(runner.run("foo").isMatched());
		assertTrue(runner.run("foobar").isMatched());
		assertFalse(runner.run("bar").isMatched());
	}
	
}

File: src/test/kotlin/com/mpe85/grampa/runner/ParseRunnerTest.kt
package com.mpe85.grampa.runner

import com.mpe85.grampa.parser.Parser
import com.mpe85.grampa.rule.Rule
import com.mpe85.grampa.rule.impl.StringRule
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class ParseRunnerTest {
  @Test
  fun test_run() {
    val parser = object : Parser<Unit> {
      override fun root(): Rule<Unit> {
        return StringRule("foo")
      }
    }
    val runner = DefaultParseRunner(parser)
    assertTrue(runner.run("foo").matched)
    assertTrue(runner.run("foobar").matched)
    assertFalse(runner.run("bar").matched)
  }
}
